EpiCommonUtils/Infrastructure/Policies/ActionMatcherPolicy.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Matching;
+
+namespace Forte.EpiCommonUtils.Infrastructure.Policies;
+
+public class ActionMatcherPolicy : MatcherPolicy, IEndpointSelectorPolicy
+{
+    public override int Order => 100;
+
+    public bool AppliesToEndpoints(IReadOnlyList<Endpoint> endpoints) => !endpoints.Any(e =>
```

Comment:
The issue is that the `Endpoint` class does not have much information about its origin - it has a display name, set of `Metadata` objects, and delegate (which points lambda inside Episerver's `MapContent` extension.
You can check that later on inside `ApplyAsync` method but then we will perform this logic on every request instead of a single time during initialization :(

Reply:
true dat. To be honest I'm not sure which option is better:
1. depend on string comparison with a class name that is in the internal namespace. 
Pros:
- we do it only once on initialization
Cons:
- we won't know when it stops working 
2. Do the check on every request
Pros:
- the logic of checking is quite solid and extensible (we can pass multiple assemblies)
Cons:
- we need to additionally pass assembly to make it work
- we need to perform the check on every request

There's no clear winner for me, maybe we should do the voting?
Having in mind that the check should not be expensive to perform on every request I would vote for (surprise, surprise 😉  ) check on every request

Reply:
I think, I would proceed also with approach to check on every request allowed assemblies, because we would have more control, whether we should check endpoints or not :)

Reply:
I really like the new approach with controller metadata